src/Featurizers/Components/VectorNormsEstimator.h

```diff
@@ -0,0 +1,181 @@
+///  \class         VectorNormsTrainingOnlyPolicy
+///  \brief         `VectorNormsEstimator` implementation details.
+///
+template <typename T, NormType NormT>
+class VectorNormsTrainingOnlyPolicy {
+public:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Types
+    // |
+    // ----------------------------------------------------------------------
+    static_assert(Traits<T>::IsVectorType, "Input type has to be vector!");
+
+    using ValueType = typename T::value_type;
+    using InputType = T;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Data
+    // |
+    // ----------------------------------------------------------------------
+    static constexpr char const * const     NameValue = VectorNormsEstimatorName;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    VectorNormsTrainingOnlyPolicy(void);
+
+    void fit(InputType const &input);
+    VectorNormsAnnotationData complete_training(void);
+
+private:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Types
+    // |
+    // ----------------------------------------------------------------------
+    std::vector<std::double_t> _norms;
+    typename TypeSelector::UpdaterTypeSelector<ValueType, NormT>::type _updater;
+};
+
+} // namespace Details
+
+/////////////////////////////////////////////////////////////////////////
+///  \typedef       VectorNormsEstimator
+///  \brief         A training-only class that finds vector-based norms of a matrix
+///
+template <
+    typename InputT,
+    NormType NormT,
+    size_t MaxNumTrainingItemsV=std::numeric_limits<size_t>::max()
+>
+using VectorNormsEstimator                       = TrainingOnlyEstimatorImpl<Details::VectorNormsTrainingOnlyPolicy<InputT, NormT>, MaxNumTrainingItemsV>;
+
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+// |
+// |  Implementation
+// |
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+
+// ----------------------------------------------------------------------
+// |
+// |  VectorNormsAnnotationData
+// |
+// ----------------------------------------------------------------------
+VectorNormsAnnotationData::VectorNormsAnnotationData(std::vector<std::double_t> norms, NormType type) :
+    Norms(std::move(norms)),
+    Type(std::move(type)) {
+        for (auto it = norms.begin(); it != norms.end(); ++it) {

```

Comment:
>for (auto it = norms.begin(); it != norms.end(); ++it) [](start = 8, length = 54)

You don't need to use an iterator here, as you are immediately dereferencing it. For improved readability, change this to for(auto const &value : norms)